webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/RuleApi.scala

```diff
@@ -933,25 +932,52 @@ class RuleApiService14 (
     }
   }
 
+  private def listCategoriesId(cat: RuleCategory) : List[RuleCategoryId] = {
+    @tailrec
+    def listCatAcc(categories: List[RuleCategory], acc: List[RuleCategoryId]) : List[RuleCategoryId] = {
+      categories match {
+        case Nil => acc
+        case c :: t => listCatAcc(t, c.id :: acc)
+      }
+    }
+    listCatAcc(cat.childs, List(RuleCategoryId("rootRuleCategory")))
+  }
+
+  // List all categories mentioned in categoryId in rules who are not in database
+  def getMissingCategories(existingCat: RuleCategory, rules: List[Rule]): List[RuleCategory] = {
+    val catIds = listCategoriesId(existingCat)
+    val rulesCat = rules.map(_.categoryId.value)
+    rulesCat
+      .filter(!catIds.map(_.value).contains(_))
+      .distinct
+      .map(s => RuleCategory(RuleCategoryId(s), s"<${s}>", s"Category ${s} has been deleted, please move rules to available categories", List(), false))
+  }
 
   def getCategoryTree(): IOResult[JRCategoriesRootEntryFull] = {
     for {
-      root  <- readRuleCategory.getRootCategory()
-      rules <- readRule.getAll()
-
+      root         <- readRuleCategory.getRootCategory()
+      rules        <- readRule.getAll()
       directiveLib <- readDirectives.getFullDirectiveLibrary()
-      groupLib <- readGroup.getFullGroupLibrary()
-      nodesLib <- readNodes.getAll()
+      groupLib     <- readGroup.getFullGroupLibrary()
+      nodesLib     <- readNodes.getAll()
       globalMode  <- getGlobalPolicyMode()
       rulesMap =  ( for {
-        rule <- rules.sortBy(_.id.serialize)
-      } yield {
-        val status = getRuleApplicationStatus(rule, groupLib, directiveLib, nodesLib, globalMode)
-        (rule, Some(status.policyMode._1), Some(status.applicationStatusDetails))
-      }).groupBy(_._1.categoryId.value)
+                    rule <- rules.sortBy(_.id.serialize)
+                  } yield {
+                    val status = getRuleApplicationStatus(rule, groupLib, directiveLib, nodesLib, globalMode)
+                    (rule, Some(status.policyMode._1), Some(status.applicationStatusDetails))
+                  }).groupBy(_._1.categoryId.value)
+      missingCatContent = getMissingCategories(root, rules.toList)
+      missingCategory   = RuleCategory(
+                            RuleCategoryId("Missing")
```

Comment:
You are defining a (public) API keyword here, since it transits out of rudder server and goes to client. 
Can you put that `Missing` in a global variable and with a comment `this Id is special and use client side to identify missing rules`.
Perhaps also use something a bit less "normal" so that it is easier to grep in case of debuging or whatever, perhaps:
`val MISSING_RULE_CAT_ID = "special-missing-rule-category"` or whatever that is likely unique?